app/src/main/java/com/example/bookclub/SearchActivity.java

```diff
@@ -153,17 +179,77 @@ public boolean onQueryTextChange(String newText) {
     public boolean onOptionsItemSelected(MenuItem item) {
         int id = item.getItemId();
         if (id == R.id.logout) {
-            Toast.makeText(SearchActivity.this, "Logout Successfully!", Toast.LENGTH_SHORT).show();
-            onLogout();
+            Toast.makeText(SearchActivity.this, getString(R.string.logout), Toast.LENGTH_SHORT).show();
+            onClickLogout();
+            return true;
+        } else if (id == R.id.action_library) {
+            Toast.makeText(SearchActivity.this, getString(R.string.in_library), Toast.LENGTH_SHORT).show();
+            onClickLibrary();
+            return true;
+        } else if (id == R.id.action_scan) {
+            Toast.makeText(SearchActivity.this, getString(R.string.scan), Toast.LENGTH_SHORT).show();
+            onClickScan();
             return true;
         }
-
         return super.onOptionsItemSelected(item);
     }
 
-    public void onLogout() {
+    private void onClickScan() {
+        //initialize intent integrator
+        IntentIntegrator intentIntegrator = new IntentIntegrator(SearchActivity.this);
+        //Set prompt text
+        intentIntegrator.setPrompt("For flash use volume up key");
+        //Set beep
+        intentIntegrator.setBeepEnabled(true);
+        //Locked orientation
+        intentIntegrator.setOrientationLocked(true);
+        //set capture activity
+        intentIntegrator.setCaptureActivity(Capture.class);
+        //initialize scan
+        intentIntegrator.initiateScan();
+    }
+
+    private void onClickLibrary() {
+        Intent intent = new Intent(this, LibraryActivity.class);
+        startActivity(intent);
+    }
+
+    public void onClickLogout() {
         ParseUser.logOut();
         Intent intent = new Intent(this, MainActivity.class);
         startActivity(intent);
     }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        //Initialize intent result
+        IntentResult intentResult = IntentIntegrator.parseActivityResult(
+                requestCode, resultCode, data
+        );
+
+        if (intentResult.getContents() != null) {
+            //when result content is not null
+            //initialize alert dialog
+            AlertDialog.Builder builder = new AlertDialog.Builder(SearchActivity.this);
+            //set title
+            builder.setTitle("Result");
+            //set message
+            builder.setMessage(intentResult.getContents());
+            //set positive button
+            builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
```

Comment:
`"OK"` should be in  string.xml since it's button text that will be shown to the user.